### PR TITLE
Refactor missing area job / move rake task to BO

### PR DESCRIPTION
**Pull request by Cruikshanks**
This covers changes to the shared gem to support creating a scheduled job in the back office, which when run will automatically update any location record with a missing EA admin area.

The first change was to refactor the logic for determining and assigning the `administrative_area` object to the `location` out of the ActiveJob and into an instance method on `location` itself.

This made it easier to create tests, and reuse the logic from a rake task.

The next step then was to move the rake task itself, previously added in PR #134, to the back office. The final intention is to run the rake task from a cron job, which we create using the [Whenever](https://github.com/javan/whenever) gem. Because we deploy the app to a multi instance environment, we don't want multiple instances all trying to do the same thing against the same record. Therefore we have logic which ensures the cron jobs are only created on a specific instance ([see Whenever's Capistrano roles support](https://github.com/javan/whenever#capistrano-roles)). In our case that is always a back office instance so it made sense to move the job to the back office project.

Finally because of our dependency on `administrative_area_lookup`, and to simplify reuse of VCR cassettes between tests, we made some changes to `spec/services/administrative_area_lookup_spec.rb`.

### CODE REVIEW

File: spec/services/administrative_area_lookup_spec.rb [hunk 1]
```diff
@@ -22,12 +22,55 @@
...
+          # the same way as the calling code from
+          # WasteExemptionsShared::ResolveAdministrativeArea does
+          response = OpenStruct.new(lookup.lookup_area_by_coordinates(coords))
+          expect(response.area_name).to eq("")
+        end
+      end
+    end
+
+    context "when the environment.data.gov.uk Web Feature Service is down" do
+      let(:coords) {
```
Comment (lewispb) [inline spec/services/administrative_area_lookup_spec.rb]: for multi line thingys I like `do..end` but up to you

File: spec/services/administrative_area_lookup_spec.rb [hunk 2]
```diff
@@ -22,12 +22,55 @@
...
+          # Need to reset the config value else the bad one gets used for subsequent tests
+          WasteExemptionsShared.config.administrative_area_api_url = "http://environment.data.gov.uk/ds/wfs"
+        end
+      end
+    end
+
+    context "when the environment.data.gov.uk Web Feature Service returns an error" do
+      let(:coords) {
+        WasteExemptionsShared::Coordinates.new(easting: 358901, northing: 171053)
+      }
```
Comment (lewispb) [inline spec/services/administrative_area_lookup_spec.rb]: seems to repeat the previous `let`. could this be DRYed?

File: spec/services/administrative_area_lookup_spec.rb (hunk 1 above)
Comment (Cruikshanks) [inline spec/services/administrative_area_lookup_spec.rb]: Yep, agreed. Will fix.

File: spec/services/administrative_area_lookup_spec.rb (hunk 2 above)
Comment (Cruikshanks) [inline spec/services/administrative_area_lookup_spec.rb]: Just not clear how I could go about it. Would moving it into a helper method be the best solution?

**Comment by Cruikshanks** [comment]
Tweaked the code as suggested @lewispb . Would you mind giving it a final pass?

File: spec/services/administrative_area_lookup_spec.rb (hunk 2 above)
Comment (lewispb) [inline spec/services/administrative_area_lookup_spec.rb]: maybe call them `somerset_coords` or wherever the coords actually are, and just have a single `let` on lines 6-7. That can then be used in any test.

**Pull request merged**